mirascope/core/base/prompt.py

```diff
@@ -421,18 +564,18 @@ async def get_base_message_params_async(
             @wraps(prompt)
             def get_base_message_params(
                 *args: _P.args, **kwargs: _P.kwargs
-            ) -> list[BaseMessageParam]:
+            ) -> Messages:
                 return parse_prompt_messages(
                     roles=["system", "user", "assistant"],
-                    template=template,
+                    template=template_or_message_function,
                     attrs=get_fn_args(prompt, args, kwargs),
                     dynamic_config=prompt(*args, **kwargs),
                 )
 
             get_base_message_params._original_fn = prompt  # pyright: ignore [reportAttributeAccessIssue,reportFunctionMemberAccess]
             return get_base_message_params
 
-    return inner
+    return inner  # pyright: ignore [reportReturnType]
```

Comment:
pyright does not consider `if` branches

Reply:
curious how this wasn't throwing an error before?

Reply:
removed